js/counties.js

```diff
@@ -5,44 +5,77 @@ function hideCounties() {
+  // into single-state files if that turns out to be useful for performance.
+  if(!countyData.has('USA')) {
+    d3.json("data/county_boundaries_USA.json", function(error, allCountiesTopo) {
+      if(error) callback(error);
+      
+      // extract the topojson to geojson
+      allCountiesGeo = topojson.feature(allCountiesTopo, allCountiesTopo.objects.counties);
+      
+      // cache in countyData
+      countyData.set('USA', allCountiesGeo);
+      
+      cacheCountyData(state, callback);
+    });
+  } else {
+    cacheCountyData(state, callback);
+  }
+}
+
+// make sure we have the state stored in countyData and then
+// call the next function (visualization)
+function cacheCountyData(state, callback) {
   // if the county boundaries for this state are already loaded, do nothing.
-  // otherwise load them now
+  // otherwise load them now. (loading currently just means subsetting them from
+  // the complete set of counties). keeping this code in here because this
+  // state-caching approach could be useful in near future
   if(!countyData.has(state)) {
     // convert state (an ID) into a file name (FIPS-based, at least for now)
     var stateFIPS = stateDict.filter(function(d) {
       return(d.state_abbv === state);
     })[0].state_FIPS;
     
-    // download the data and run the processing function
-    oneStateCounties = d3.json("data/" + stateFIPS + "-quantized.json", function(error, oneStateCounties) {
-      if(error) callback(error);
-      countyData.set(state, oneStateCounties);
-      callback(error, countyData.get(state));
+    // subset the data and run the processing function
+    oneStateCounties = countyData.get('USA').features.filter(function(d) {
+      return(d.properties.STATEFP === stateFIPS)
     });
+    countyData.set(state, oneStateCounties);
+    callback(null, countyData.get(state));
+    
+    // here's how we used to download the data and run the processing function:
+    // oneStateCounties = d3.json("data/" + stateFIPS + "-quantized.json", function(error, oneStateCounties) {
+    //   if(error) callback(error);
+    //   countyData.set(state, oneStateCounties);
+    //   callback(error, countyData.get(state));
+    // });
+    
   } else {
     // if we already have the data, just run the processing function
     callback(null, countyData.get(state));
   }
 }
 
+// visualize
 function displayCountyData(error, activeCountyData) {
     if(error) throw error;
     
-    // extract the topojson to geojson
-    var geojson = topojson.feature(activeCountyData, activeCountyData.objects.state);
-    var yeargeo = geojson.features.filter(function(d) {
-      return d.properties.year == 2015;
-    });
-  
     // create paths
     var countyBounds = map.selectAll(".county")
-      .data(yeargeo, function(d) {
-        return d.properties.county_FIPS;
+      .data(activeCountyData, function(d) {
+        return d.properties.GEOID;
```

Comment:
thinking ahead, is this where we would easily drop or add circles when zoomed into a state level?

Reply:
yes, i think that's right...